src/uwtools/drivers/forecast.py

```diff
@@ -138,82 +132,60 @@ def requirements(self) -> None:
         ???
         """
 
-    def resources(self, platform: dict) -> Mapping:
+    def resources(self) -> Mapping:
         """
         Parses the config and returns a formatted dictionary for the batch script.
         """
-        # Add required fields to platform.
-        # Currently supporting only slurm scheduler.
+
         return {
-            "account": platform["account"],
-            "nodes": 1,
-            "queue": "batch",
-            "scheduler": self.job_scheduler,
-            "tasks_per_node": 1,
-            "walltime": "00:01:00",
+            "account": self._experiment_config["user"]["account"],
+            "scheduler": self._experiment_config["platform"]["scheduler"],
+            **self._config["jobinfo"],
         }
 
-    def run(self) -> None:
+    def run(self, cycle: datetime) -> None:
         """
         Runs FV3 either as a subprocess or by submitting a batch script.
         """
-        # Read in the config file.
-        forecast_config = self._config["forecast"]
-        platform_config = self._config["platform"]
-
         # Prepare directories.
-        run_directory = forecast_config["RUN_DIRECTORY"]
+        run_directory = self._config["run_dir"]
         self.create_directory_structure(run_directory, "delete")
 
-        static_files = forecast_config["STATIC"]
-        self.stage_files(run_directory, static_files, link_files=True)
-        cycledep_files = forecast_config["CYCLEDEP"]
-        self.stage_files(run_directory, cycledep_files, link_files=True)
-
-        # Create the job script.
-        platform_resources = self.resources(platform_config)
-        batch_script = self.batch_script(platform_resources)
-        args = "--export=None"
-        run_command = self.run_cmd(
-            args,
-            run_cmd=platform_config["MPICMD"],
-            exec_name=forecast_config["EXEC_NAME"],
-        )
-        batch_script.append(run_command)
+        self._config["cycle-dependent"].update(self._define_boundary_files())
 
-        if self._dry_run:
-            # Apply switch to allow user to view the run command of config.
-            # This will not run the job.
-            logging.info("Batch Script:")
-            logging.info(batch_script)
-            return
+        for file_category in ["static", "cycle-dependent"]:
+            self.stage_files(run_directory, self._config[file_category], link_files=True)
+
+        run_time_args = self._config["runtime_info"].get("mpi_args", [])
 
-        # Run the job.
         if self._batch_script is not None:
+            batch_script = self.batch_script()
+
+            if self._dry_run:
+                # Apply switch to allow user to view the run command of config.
+                # This will not run the job.
+                logging.info("Batch Script:")
+                logging.info(batch_script)
```

Comment:
In general, I agree with that. In this case, though, I'm curious if it gets in the way of piping the output to a file if that's how the user is working with the tool.

Reply:
Logging goes to `stderr` and users should never expect logging to be usable programmatically. It's for humans to read, not programs.

Data goes to `stdout` and is what users should rely on. It will never include logging timestamps, etc., and is meant to be used programmatically.

Reply:
If the intention here is for `batch_script` to be usable programmatically, we should `print()` or `f.write()` it to `stdout`, not log it.